app/src/views/home/home_screen.tsx

```diff
@@ -73,6 +76,15 @@ export default function HomeScreen({
     outputRange: [height * 0.18, 0],
   });
 
+  // Handle restricted navigation for guest users
+  const handleRestrictedAccess = (screenName: string) => {
+    if (userIsGuest) {
+      setShowGuestPopup(screenName);
+    } else {
+      navigation.navigate(screenName);
+    }
+  };
+
```

Comment:
That is a clever way to centralize navigations if authorized !

Reply:
Thank you !